src/main/java/io/cdap/plugin/google/common/GoogleAuthBaseConfig.java

```diff
@@ -229,27 +243,48 @@ private boolean validateServiceAccount(FailureCollector collector) {
     return collector.getValidationFailures().size() == 0;
   }
 
-  private void validateCredentials(FailureCollector collector, GoogleDriveClient driveClient) throws IOException {
-    try {
-      driveClient.checkRootFolder();
-    } catch (GoogleJsonResponseException e) {
-      collector.addFailure(e.getDetails().getMessage(), "Provide valid credentials.")
-        .withConfigProperty(NAME_SERVICE_ACCOUNT_TYPE)
-        .withStacktrace(e.getStackTrace());
-    }
-  }
-
-  private void validateDirectoryIdentifier(FailureCollector collector, GoogleDriveClient driveClient)
+  private boolean isDirectoryOrFileAccessible(FailureCollector collector, GoogleDriveClient driveClient)
     throws IOException {
-    if (!containsMacro(DIRECTORY_IDENTIFIER)) {
+
+    if (getIdentifierType().equals(IdentifierType.DIRECTORY_IDENTIFIER) &&
+      directoryIdentifier != null && !containsMacro(DIRECTORY_IDENTIFIER)) {
       try {
         driveClient.isFolderAccessible(directoryIdentifier);
+        return true;
       } catch (GoogleJsonResponseException e) {
         collector.addFailure(e.getDetails().getMessage(), "Provide an existing folder identifier.")
           .withConfigProperty(DIRECTORY_IDENTIFIER)
           .withStacktrace(e.getStackTrace());
       }
     }
+
+    if (getIdentifierType().equals(IdentifierType.FILE_IDENTIFIER) &&
```

Comment:
Generally we use constants first to avoid `NPE`,

`IdentifierType.FILE_IDENTIFIER.equals(getIdentifierType())`

Reply:
getIdentifierType can never be null as null is already handled there

Reply:
It is a recommended Java practice.

Reply:
changed